src/main/java/org/jabref/logic/openoffice/backend/Codec52.java

```diff
@@ -0,0 +1,137 @@
+        public final CitationType citationType;
+        /** Citation keys embedded in the reference mark. */
+        public final List<String> citationKeys;
+
+        ParsedMarkName(String i, CitationType citationType, List<String> citationKeys) {
+            Objects.requireNonNull(i);
+            Objects.requireNonNull(citationKeys);
+            this.i = i;
+            this.citationType = citationType;
+            this.citationKeys = citationKeys;
+        }
+    }
+
+    /**
+     * Integer representation was written into the document in JabRef52, keep it for compatibility.
+     */
+    private static CitationType citationTypeFromInt(int i) {
+        return switch (i) {
+            case 1 -> CitationType.AUTHORYEAR_PAR;
+            case 2 -> CitationType.AUTHORYEAR_INTEXT;
+            case 3 -> CitationType.INVISIBLE_CIT;
+            default -> throw new IllegalArgumentException("Invalid CitationType code");
+        };
+    }
+
+    private static int citationTypeToInt(CitationType i) {
+        return switch (i) {
+            case AUTHORYEAR_PAR -> 1;
+            case AUTHORYEAR_INTEXT -> 2;
+            case INVISIBLE_CIT -> 3;
+            default -> throw new IllegalArgumentException("Invalid CitationType");
+        };
+    }
+
+    /**
+     * Produce a reference mark name for JabRef for the given citationType and list citation keys that
+     * does not yet appear among the reference marks of the document.
+     *
+     * @param usedNames    Reference mark names already in use.
+     * @param citationKeys Identifies the cited sources.
+     * @param citationType Encodes the effect of withText and inParenthesis options.
+     *
+     * The first occurrence of citationKeys gets no serial number, the second gets 0, the third 1 ...
+     *
+     * Or the first unused in this series, after removals.
+     */
+    public static String getUniqueMarkName(Set<String> usedNames,
+                                           List<String> citationKeys,
+                                           CitationType citationType) {
+
+        String citationKeysPart = String.join(",", citationKeys);
+
+        int i = 0;
+        int citTypeCode = citationTypeToInt(citationType);
+        String name = BIB_CITATION + '_' + citTypeCode + '_' + citationKeysPart;
+        while (usedNames.contains(name)) {
+            name = BIB_CITATION + i + '_' + citTypeCode + '_' + citationKeysPart;
+            i++;
+        }
+        return name;
+    }
+
+    /**
+     * Parse a JabRef (reference) mark name.
+     *
+     * @return Optional.empty() on failure.
+     *
+     */
+    public static Optional<ParsedMarkName> parseMarkName(String refMarkName) {
+
+        Matcher citeMatcher = CITE_PATTERN.matcher(refMarkName);
+        if (!citeMatcher.find()) {
+            return Optional.empty();
+        }
+
+        List<String> keys = Arrays.asList(citeMatcher.group(3).split(","));
+        String i = citeMatcher.group(1);
```

Comment:
i is normally used only in loops, but for the moment it doesn't matter. You can fix this in a later PR.